src/engine/strat_engine/backstore/range_alloc.rs

```diff
@@ -409,6 +409,13 @@ impl RangeAllocator {
             .union(&segs)
             .expect("all segments verified to be in available ranges");
     }
+
+    /// Increase the available size of the RangeAlloc data structure.
+    ///
+    /// Precondition: new_size > self.limit
+    pub fn increase_size(&mut self, new_size: Sectors) {
```

Comment:
I think it makes sense to add an assertion ```new_size >= self.segments.limit``` here. Cheap, and definitely necessary.

Reply:
Okay, that sounds good to me. It's consistent with what we usually do when we add a precondition.